lang/wasm_alien_thread_runner.hh

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <functional>
+#include <mutex>
+#include <queue>
+#include <condition_variable>
+#include <thread>
+
+#include <seastar/core/future.hh>
+
+#include "rust/cxx.h"
+#include "rust/wasmtime_bindings.hh"
+
+namespace wasm {
+
+struct wasm_compile_task {
+    std::function<void()> func;
+    seastar::promise<rust::Box<wasmtime::Module>>& done;
+    int shard;
+};
+
+struct task_queue {
+    std::mutex _mut;
+    std::condition_variable _cv;
+    static constexpr unsigned QUEUE_SIZE = 128;
+    std::queue<std::optional<wasm_compile_task>> _pending;
+public:
+    std::optional<wasm_compile_task> pop_front();
+    void push_back(std::optional<wasm_compile_task> work_item);
+    size_t size() { return _pending.size(); }
+};
+
+class alien_thread_runner {
+    std::thread _thread;
+    task_queue _pending_queue;
```

Comment:
probably need to put `_pending_queue` before `_thread`, so `_pending_queue` is initialized before `_thread`, as in the constructor of `alien_thread_runner`, it initializes `_thread` using `_pending_queue`.

Reply:
Good catch.